### PR TITLE
Hide python options in `uv tool list` help

### LINKED ISSUE #5982: Hide Python options in `uv tool list` help
We don't need `--python-preference` and `--no-python-downloads` here.

e.g.

```
❯ uv tool list --help
List installed tools

Usage: uv tool list [OPTIONS]

Options:
      --show-paths  Whether to display the path to each tool environment and installed executable

Cache options:
  -n, --no-cache               Avoid reading from or writing to the cache, instead using a temporary directory for the duration of the
                               operation [env: UV_NO_CACHE=]
      --cache-dir <CACHE_DIR>  Path to the cache directory [env: UV_CACHE_DIR=]

Python options:
      --python-preference <PYTHON_PREFERENCE>  Whether to prefer uv-managed or system Python installations [possible values:
                                               only-managed, managed, system, only-system]
      --python-fetch <PYTHON_FETCH>            Whether to automatically download Python when required [possible values: automatic,
                                               manual]

...
```

### PR DESCRIPTION
**Pull request by eth3lbert**
## Summary

Closes #5982 .

## Test Plan

```
cargo run tool list --help
```

### CODE REVIEW

**Comment by eth3lbert** [comment]
```
List installed tools

Usage: uv tool list [OPTIONS]

Options:
      --show-paths  Whether to display the path to each tool environment and installed executable

Cache options:
  -n, --no-cache               Avoid reading from or writing to the cache, instead using a temporary
                               directory for the duration of the operation [env: UV_NO_CACHE=]
      --cache-dir <CACHE_DIR>  Path to the cache directory [env: UV_CACHE_DIR=]

Global options:
  -q, --quiet                      Do not print any output
  -v, --verbose...                 Use verbose output
      --color <COLOR_CHOICE>       Control colors in output [default: auto] [possible values: auto,
                                   always, never]
      --native-tls                 Whether to load TLS certificates from the platform's native
                                   certificate store [env: UV_NATIVE_TLS=]
      --offline                    Disable network access
      --no-progress                Hide all progress outputs
      --config-file <CONFIG_FILE>  The path to a `uv.toml` file to use for configuration [env:
                                   UV_CONFIG_FILE=]
      --no-config                  Avoid discovering configuration files (`pyproject.toml`,
                                   `uv.toml`) [env: UV_NO_CONFIG=]
  -h, --help                       Display the concise help for this command
  -V, --version                    Display the uv version
```

**Comment by eth3lbert** [comment]
Should we raise an "unexpected argument" error when these arguments are passed?

File: crates/uv-cli/src/lib.rs [hunk 1]
```diff
@@ -2771,6 +2771,12 @@ pub struct ToolListArgs {
     /// Whether to display the path to each tool environment and installed executable.
     #[arg(long)]
     pub show_paths: bool,
+
+    // Hide unsupported python options.
```
Comment (zanieb) [inline crates/uv-cli/src/lib.rs:2775]:
```suggestion
    // Hide unused global Python options.
```

**Comment by zanieb** [comment]
No it needs to be okay to provide them, since they're global options.

**Review by zanieb — APPROVED** [review]

**Pull request merged**